### PR TITLE
Add bundle

### PR DESCRIPTION
**Pull request by junmin-kim**
Add tizen bundle and bundle test example

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [4ae0539cbe915a1e00eba591c59de08417a58f13] - Code Rule Check Result:
framework/include/tizen/bundle.h:113: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:120: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:162: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:162: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:163: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:163: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:164: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:164: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:187: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:187: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:188: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:188: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:189: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:189: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:212: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:212: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:213: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:213: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:214: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:214: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:236: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:237: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:237: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:238: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:238: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:239: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:239: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:240: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:240: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:270: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:270: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:271: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:271: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:272: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:272: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:273: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:273: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:504: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:639: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:639: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:640: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:640: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:641: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:641: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:667: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:667: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:668: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:668: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:669: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:669: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:703: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:703: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:704: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:704: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:705: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:705: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:732: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:732: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:733: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:733: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:734: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:734: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:765: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:765: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:766: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:766: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:767: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:767: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:799: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:799: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:800: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:800: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:801: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:801: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:802: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:802: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:830: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:830: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:831: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:831: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:832: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:832: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:833: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:833: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:862: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:862: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:863: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:863: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:864: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:864: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/tizen/bundle.h:865: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/tizen/bundle.h:865: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/tizen/bundle/bundle.c:796: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [4ae0539cbe915a1e00eba591c59de08417a58f13] - Code Rule Check (C++) OK.

**Comment by junmin-kim** [comment]
@Taejun-Kwon We need a discussion about where the bundle is located.

**Comment by seinfra** [comment]
Target : [72798f9c6dee8716b51c5077d4bf68f5cd7e82ec] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [72798f9c6dee8716b51c5077d4bf68f5cd7e82ec] - Code Rule Check Result:
framework/include/app/bundle.h:113: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:120: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:163: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:163: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:188: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:188: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:236: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:504: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:640: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:640: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:668: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:668: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:704: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:704: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:733: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:733: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/app/bundle.h:766: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/app/bundle.h:766: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/app/bundle/bundle.c:796: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [b16e2462972310da6c8565835ee062f355b90d9e] - Code Rule Check OK.

**Comment by juitem** [comment]
As this is the first commit, would you please add one simple comment describes the bundle?

**Comment by seinfra** [comment]
Target : [b16e2462972310da6c8565835ee062f355b90d9e] - Code Rule Check (C++) OK.

**Comment by junmin-kim** [comment]
> As this is the first commit, would you please add one simple comment describes the bundle?

I added a description of the data bundle.

**Comment by seinfra** [comment]
Target : [3bce88a981f2ab788b4a8252ce026c4afdb494ec] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3bce88a981f2ab788b4a8252ce026c4afdb494ec] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [778a12cb571816775da6e1dd33b4df9c7a779c13] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [778a12cb571816775da6e1dd33b4df9c7a779c13] - Code Rule Check (C++) OK.

File: apps/examples/bundle_test/Kconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config EXAMPLES_BUNDLE_TEST
+	bool "Bundle Test Example"
+	default n
```
Comment (jeongarmy) [inline apps/examples/bundle_test/Kconfig]: I think it should have a dependency on CONFIG_TIZEN_BUNDLE

**Comment by seinfra** [comment]
Target : [d4ef18bd77d616a07232b0b1d791dd7d119cf459] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d4ef18bd77d616a07232b0b1d791dd7d119cf459] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b48f1355360567377c22b1ffb4622bf90595b087] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b48f1355360567377c22b1ffb4622bf90595b087] - Code Rule Check (C++) OK.

File: apps/examples/bundle_test/Kconfig (hunk 1 above)
Comment (junmin-kim) [reply apps/examples/bundle_test/Kconfig]: Thanks for the review. I'll add it.

**Comment by seinfra** [comment]
Target : [2f97fc564c6bbf58054a69fe2bdeaedc687c5519] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2f97fc564c6bbf58054a69fe2bdeaedc687c5519] - Code Rule Check (C++) OK.

File: framework/include/app/bundle.h [hunk 2]
```diff
@@ -0,0 +1,896 @@
+/*
```
Comment (zhouxinhe) [inline framework/include/app/bundle.h]: Please check mode of files. ( bundle.h and bundle/\*.\* )

**Comment by jeongchanKim** [comment]
I already ported the bundle in sec github. How about back-porting from there?
If not, we'd better modify the sec github also.

File: framework/src/app/bundle/bundle_util.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
```
Comment (jeongchanKim) [inline framework/src/app/bundle/bundle_util.c]: Please check the license year.

File: apps/examples/bundle_test/Kconfig (hunk 1 above)
Comment (jeongchanKim) [reply apps/examples/bundle_test/Kconfig]: I'm not sure it should have dependency on  CONFIG_TIZEN_BUNDLE.
It is not only for TIZEN.

File: apps/examples/bundle_test/bundle_test_main.c [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+	bundle_free(b1);
+	bundle_free(b2);
+	printf("test_bundle_dup done\n");
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int bundle_test_main(int argc, char *argv[])
```
Comment (jeongchanKim) [inline apps/examples/bundle_test/bundle_test_main.c]: I think you'd better change these to testcase for supporting bundles to public API.

**Pull request closed without merging**